modules/photon/system-part1/linker.ld

```diff
@@ -3,7 +3,7 @@ MEMORY
     APP_FLASH (rx)  : ORIGIN = 0x08020000, LENGTH = 256K
 
     /* todo - SRAM must start also at an offset after what has been reserved for system-module 1 */
-    SRAM      (rwx) : ORIGIN = 0x20000000, LENGTH = 384
+    SRAM      (rwx) : ORIGIN = 0x20000000, LENGTH = 1K
```

Comment:
oh, so in case of Photon I can extend part1's SRAM region only up to 0x20000300 address ([system_part1_module_ram_end](https://github.com/spark/firmware/blob/develop/modules/photon/system-part1/module_system_part1_export.ld#L7)), which is 768 bytes? This is still enough to get this branch compiled with gcc 5.4.1 as is -- just want to ensure this PR results in correct firmware until I investigate what causes increase in static memory usage.